app/src/main/java/com/nervousfish/nervousfish/activities/KeyManagementActivity.java

```diff
@@ -17,6 +45,67 @@
     @Override
     protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_key_management);
+        this.setContentView(R.layout.activity_key_management);
+        final IServiceLocator serviceLocator = NervousFish.getServiceLocator();
+        final IDatabase database = serviceLocator.getDatabase();
+        String title = "";
+        String[] key = {};
+        try {
+            this.myProfile = database.getProfiles().get(0);
+            title = this.myProfile.getKeyPairs().get(0).getName();
+            key = this.myProfile.getKeyPairs().get(0).getPublicKey().getKey().split(" ");
+
+        } catch (IOException e) {
+            LOGGER.error("Could not get my public key from the database ", e);
+        }
+
+        final ArrayList<IKey> list = new ArrayList<>();
+        for (final KeyPair kp : this.myProfile.getKeyPairs()) {
+            list.add(kp.getPublicKey());
+        }
+
+        ListviewActivityHelper.setKeys(this, list, R.id.list_view_edit_keys);
+
+        final ListView lv = (ListView) this.findViewById(R.id.list_view_edit_keys);
+
+        final AlertDialog.Builder builder = new AlertDialog.Builder(this);
```

Comment:
Because with @jverbraeken 's refactor suggestion I can't fit it in the `MyOnItemClickListener` completely. Now I've scrapped the static keyword and moved as much code as I could into the `MyOnItemClickListener`  but it's still not all the code.